### PR TITLE
Add Automatic OmniAuth

### PR DESCRIPTION
**Pull request by paulcsmith**
* Saves the user's credentials (ID from Automatic and the OAuth token)
* Redirects without saving credentials if the user's credentials are already saved

In the next PR:
* Redirect to driver data if the user is already signed in
* Redirect to sign in page if the user is not logged in

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -12,6 +12,8 @@ gem "flutie"
 gem "high_voltage"
 gem "inline_svg"
 gem "jquery-rails"
+gem "omniauth", "~> 1.2"
```
Comment (graysonwright) [inline Gemfile]: It looks like `omniauth` is a dependency of `omniauth-automatic`. Do we need this line?

File: README.md [hunk 2]
```diff
@@ -22,6 +22,18 @@ is [purposefully excluded from the project's `Gemfile`][exclude].
...
+
+You will need to sign up/in at the [Automatic Developer
+page](https://developer.automatic.com/sign-in/)
+
+Once you have an account, copy the client ID and client secret to your `.env`
+file. You will also need to change the Redirect URL in the Automatic developer
+portal to `http://localhost:3000/auth/automatic/callback`.
+
+You may need to change the port if your local server is started on a different
+port.
```
Comment (graysonwright) [inline README.md]: :clap: great docs.

File: app/views/driver_data/index.html.erb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div class="container">
+<div class="container" data-role="archetype-container">
```
Comment (graysonwright) [inline app/views/driver_data/index.html.erb]: This container contains both the archetype and the stat chips. What's the motivation for this change?

Comment (graysonwright) [inline app/views/driver_data/index.html.erb]: ah, looks like a test hook. Thoughts on moving this `data-role` attribute into the `archetype` partial?

File: app/views/sessions/new.html.erb [hunk 4]
```diff
@@ -11,7 +11,10 @@
     </div>
 
     <div class="auth-actions">
-      <a href="#" class="button auth-button dummy">Sign in with Automatic</a>
+      <%= link_to "Sign in with Automatic",
+        "/auth/automatic",
+        class: "button auth-button",
+        "data-role" => "sign-in-button" %>
```
Comment (graysonwright) [inline app/views/sessions/new.html.erb]: I believe we can set data attributes by passing in a `data: { role: "sign-in-button" }` argument. Thoughts on that approach?

File: app/views/driver_data/index.html.erb (hunk 3 above)
Comment (paulcsmith) [inline app/views/driver_data/index.html.erb]: Or maybe change the data-role to `"driver-data-container"` since that's really what I care about, not necessarily the archetype or the stats

File: app/views/sessions/new.html.erb (hunk 4 above)
Comment (paulcsmith) [inline app/views/sessions/new.html.erb]: My personal preference is for the current approach. I think it's cleaner and easier to read, but if you feel strongly I don't mind changing it

File: spec/features/guest_visits_homepage_spec.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      expect(User.all).to eq [user]
+      expect(page).to have_archetype
+    end
+  end
+
+  def access_token
+    "my_access_token"
+  end
+
+  def have_archetype
```
Comment (graysonwright) [inline spec/features/guest_visits_homepage_spec.rb]: :clap:

File: app/views/sessions/new.html.erb (hunk 4 above)
Comment (graysonwright) [inline app/views/sessions/new.html.erb]: nope, don't feel strongly. let's leave it.

File: README.md (hunk 2 above)
Comment (paulcsmith) [inline README.md]: Thanks :)

File: app/views/driver_data/index.html.erb (hunk 3 above)
Comment (graysonwright) [inline app/views/driver_data/index.html.erb]: :+1:

File: Gemfile (hunk 1 above)
Comment (paulcsmith) [inline Gemfile]: Good catch!

**Comment by graysonwright** [comment]
Couple questions, but overall looks great! Ship when you're ready.

**Comment by paulcsmith** [comment]
Thanks for the review!
